app/views/courses/create.json.jbuilder

```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
 json.course do
-  json.call(@course, :title, :description,
-            :start, :end, :school, :term, :subject, :slug)
+  if @course
+    json.call(@course, :title, :description,
+              :start, :end, :school, :term, :subject, :slug)
+  end
```

Comment:
I do understand why you do this, but I would prefer not to change what the course creation endpoint renders at all, as it is difficult to be 100% sure that this will not cause any problems.

I think one option is to make `CourseCreationManager.create` method keep returning a course even though persistence to the database fails (as it's working now), but skipping the side effect actions. We can still determine if the course was created based on the presence of  `@course.id`.
It would be helpful  to add some comments to the `CourseCreationManager.create` method, clarifying that it returns either a newly created course object or a course object with `nil` for the `id` if creation fails.

I understand if you think this proposed solution seems less clean, but I think the key point is to minimize changes.

Reply:
@gabina Ok, I've changed it back to returning the course no matter what happened. Added a comment too.

Reply:
Thanks!
I think it’s a good idea to keep the premature `return` you had included in `CoursesController.create` if `@course` was not saved to skip calling `update_courses_wikis`, `update_course_wiki_namespaces`, etc. You can check if @course.id is nil or not to determine this. Do you mind adding it? This should be ready to go after that.

Reply:
@gabina I don't see where you would return early, I thought we are still returning the course no matter if the save succeeded or not.

Reply:
hmm I cannot see your previous commits because you force-pushed. Right now the only change is in `CourseCreationManager.create`. That method is called in `CoursesController.create`. I think we could follow your previous proposal to return earlier if the course was not persisted to the database correctly and make something like:

```
  def create
    require_signed_in

    course_creation_manager = CourseCreationManager.new(course_params, wiki_params,
                                                        params[:course][:scoping_methods],
                                                        initial_campaign_params,
                                                        instructor_role_description, current_user,
                                                        params[:course][:ta_support])
    unless course_creation_manager.valid?
      render json: { message: course_creation_manager.invalid_reason },
             status: :not_found
      return
    end
    @course = course_creation_manager.create
    # return early if the course was not persisted to the db
    return if @course.id.nil?
    update_courses_wikis
    update_course_wiki_namespaces
    update_academic_system
    update_course_format
  end
```

I didn't test it, but I think that doing something like that would make sense. WDYT?

Reply:
@gabina Yes, that makes sense. I've added that change to the PR.